docs/design/storage_abstraction.md

```diff
@@ -0,0 +1,275 @@
+
+```move
+module moveos_std::type_table {
+
+    #[private_generics(T)]
+    public fun add<V: key>(table: &mut TypeTable, val: V);
+
+    #[private_generics(T)]
+    public fun borrow<V: key>(table: &TypeTable): &V;
+
+    #[private_generics(T)]
+    public fun borrow_mut<V: key>(table: &mut TypeTable): &mut V;
+
+    #[private_generics(T)]
+    public fun remove<V: key>(table: &mut TypeTable): V;
+
+    #[private_generics(T)]
+    public fun contains<V: key>(table: &TypeTable): bool;
+}
+```
+
+The methods in TypeTable rely on the type safety guarantee of `private_generics`, with constraints similar to Move's global storage instructions.
+
+### AccountStorage
+
+AccountStorage is an abstraction of a user's storage space in Move, which contains two Tables, the Resource Table and Module Table, allowing Resources and Modules to be directly manipulated in Move without the need for global storage instructions.
+
+```move
+struct AccountStorage has key {
+    resources: TypeTable,
+    modules: Table<String, vector<u8>>,
+}
+```
+
+It mainly provides the following API:
+
+```
+module moveos_std::account_storage{
+
+    #[private_generics(T)]
+    /// Move a resource to the account's storage
+    /// This function equates to `move_to<T>(&signer, resource)` instruction in Move
+    public fun global_move_to<T: key>(ctx: &mut StorageContext, account: &signer, resource: T);
+
+    #[private_generics(T)]
+    /// Move a resource from the account's storage
+    /// This function equates to `move_from<T>(address)` instruction in Move
+    public fun global_move_from<T: key>(ctx: &mut StorageContext, account: address): T;
+
+    #[private_generics(T)]
+    /// Check if the account has a resource of the given type
+    /// This function equates to `exists<T>(address)` instruction in Move
+    public fun global_exists<T: key>(ctx: &StorageContext, account: address) : bool;
+
+    /// Check if the account has a module with the given name
+    public fun exists_module(ctx: &StorageContext, account: address, name: String): bool;
+
+    /// Publish modules to the account's storage
+    public fun publish_modules(ctx: &mut StorageContext, account: &signer, modules: vector<vector<u8>>);
+}
+```
+
+The above methods can replace the global storage instructions in Move, so that all storage-related operations in Move are ultimately unified to Operations on Tables. 
+
+At the same time,this approach also solves a reference problem in Move whereby developers cannot return a reference borrowed through the `borrow_global<T>` instruction in their methods, since this reference is created out of thin air and cannot be returned as a method return value. However, if the reference is borrowed through `StorageContext`, then that goal can be achieved, and the lifetime of the reference is bound to the lifetime of `StorageContext`. This eliminates the need for developers to manually track reference lifetimes and ensures the safety of their usage in the Move program.
```

Comment:
So does it mean I don't need to use the `acquires` annotation anymore? and is it safe?
https://move-language.github.io/move/global-storage-operators.html#reference-safety-for-global-resources

Reply:
It is a good question. We don't need the `acquires`, but the safety need to be re-evaluated. I record an issue #82